src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/hasNoAmpSupport.js

```diff
@@ -0,0 +1,4 @@
+export default classification =>
+  !!(
+    classification === 'vj-amp-not-supported' || classification === 'idt1-amp'
+  );
```

Comment:
Can we make use of the `ampSupported` function in `src/app/routes/cpsAsset/getInitialData/convertToOptimoBlocks/blocks/include/includeClassifier.js`?

If we decide to keep this module, we can remove the double NOT operators as the equality comparisons will always return a boolean value.

Reply:
We could, although from the looks of it that's only used to check for amp support on VJ atm. If we export that function and reuse that means we'll be doing another query on url again. Or we could actually add the `hasAmpSupport` determination within the classifier, just depends if people prefer keeping it separate.

Reply:
I think the logic looks fine here (minus the double negation as @simonsinclair says), we don't need to query url again, that complexity has been encapsulated in the `includeClassifier`. 

I'd probably say there isn't enough logic here to warrant seperating out into a seperate module but it is always a judgement call. With seperation of concerns, excessive seperation harms readability whilst not enough has the same problem, it's all a balance! One guideline I've heard many times is if your logic spans 3 lines + that tends to indicate it's worth extracting out a module, that again though is just a _guideline_.

Reply:
I think you're right to keep them separate:

- The classifier's job is to put an Include into a bucket. I would argue that the `ampSupported` function within `includeClassifier` should be be `hasAmpVersion`.
- Then we can then decide which buckets we want to render on AMP and which not. This could be achieved with a function like yours `isAmpSupported(classification)`.

Reply:
I decided to leave the logic in separate module. Mainly because of the number of test cases we have to run through, and the main test file for the include pre-processor is pretty large already. Also it's easy to add edge cases.